src/hexdoc/patchouli/page/pages.py

```diff
@@ -47,13 +53,37 @@ class EmptyPage(Page, type="patchouli:empty", template_type="patchouli:page"):
 
 
 class EntityPage(PageWithText, type="patchouli:entity"):
+    _default_name: LocalizedStr = PrivateAttr()
+    _texture: PNGTexture = PrivateAttr()
+
     entity: Entity
     scale: float = 1
     offset: float = 0
     rotate: bool = True
     default_rotation: float = -45
     name: LocalizedStr | None = None
 
+    def get_name(self):
+        if self.name is None:
```

Comment:
According to the [docs](https://vazkiimods.github.io/Patchouli/docs/patchouli-basics/page-types/#name-string), the entity's name is used when `name` is missing or empty, so maybe `if self.name is None or not self.name.value`? Or alternatively we could add a before-validator to the `name` field that replaces empty strings with `None`.